### PR TITLE
Add Flags.parseCurrentPackage support

### PR DESCRIPTION
**Pull request by val-fed**

### CODE REVIEW

File: common/flags/Flags.java [hunk 1]
```diff
@@ -79,6 +80,24 @@
     return instance._parse(args);
   }
 
+  /**
+   * Initializes flag values from command-line style arguments. Scanners all flags of the current
```
Comment (oferb) [inline common/flags/Flags.java]: s/Scanners/Scans

File: common/flags/Flags.java [hunk 2]
```diff
@@ -79,6 +80,24 @@
...
+   * package.
+   *
+   * @param args command-line arguments to parse values from
+   */
+  public static String[] parseCurrentPackage(String[] args) {
+    // getting the caller class name
+    String className = Thread.currentThread().getStackTrace()[2].getClassName();
```
Comment (oferb) [inline common/flags/Flags.java]: Can you document why we need index 2?

File: common/flags/Flags.java [hunk 3]
```diff
@@ -79,6 +80,24 @@
     return instance._parse(args);
   }
 
+  /**
+   * Initializes flag values from command-line style arguments. Scanners all flags of the current
+   * package.
+   *
+   * @param args command-line arguments to parse values from
+   */
+  public static String[] parseCurrentPackage(String[] args) {
```
[the remark this answers is not in the document]
Comment (val-fed) [reply common/flags/Flags.java]: - [x] Done

File: common/flags/Flags.java [hunk 4]
```diff
@@ -79,6 +80,24 @@
...
+    // getting the caller class name
+    String className = Thread.currentThread().getStackTrace()[2].getClassName();
+    try {
+      instance()
+          .scanPackages(Collections.singletonList(Class.forName(className).getPackage().getName()));
+    } catch (ClassNotFoundException e) {
```
[the remark this answers is not in the document]
Comment (val-fed) [reply common/flags/Flags.java]: - [x] Done

File: common/flags/Flags.java (hunk 1 above)
Comment (val-fed) [reply common/flags/Flags.java]: - [x] Done

File: common/flags/Flags.java (hunk 2 above)
Comment (val-fed) [reply common/flags/Flags.java]: - [x] Done

**Pull request merged**